backend/src/main/java/com/google/sps/servlets/LoadUserPreferencesServlet.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.User;
+import com.google.sps.datastore.UserDatastore;
+import java.io.IOException;
+import javax.annotation.Nullable;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;
+
+/** Servlet that returns a JSON object of the user's preferences */
+@WebServlet("/api/v1/load-user")
+public class LoadUserPreferencesServlet extends HttpServlet {
```

Comment:
nit: (and more like a question for Java servlet) It looks like the static strings here and the member function getUsername() are similar to AddParticipant servlet. Can we inherit that servlet and only override doGet() function? (And much of the doGet() function can be of its own function in AddParticipant, so this class can be just calling that function plus the return of userJson)

Reply:
WDYT about adding a new class called GetUsername.java or something that has the one getUsername() method, which all the servlets can call? so not tying that method directly to only AddParticipantServlet, since that seems a bit arbitrary.

Reply:
Delegation to a helper class could work. Another option would be a static utility class that takes an email String and returns the username

Reply:
Sure help class wfm. My comment was more like seeing a lot repeated code, including the private variables, etc. and trying to think of ways to optimize. If the helper is not used by other servlets (or is it?), it's the same with or without helper?